### PR TITLE
infra: delete obsolete docker images and scripts

### PR DESCRIPTION
**Pull request by geekflyer**
This deletes obsolete files in the docker directory

### Test Plan
CI/CD passes. Also did a "Find" for each file that I deleted to see where/if it's being used.

### CODE REVIEW

File: docker/README.md [hunk 1]
```diff
@@ -1,8 +1,5 @@
 # Docker
 
-This directory contains [Docker](https://www.docker.com/) configuration and scripts for building
-images of the validator node, faucet server, and client.
-
-Each subdirectory contains a build.sh to build the respective Docker container.
+This directory contains [Docker](https://www.docker.com/) configuration for building Aptos docker images.
```
Comment (sherry-x) [inline docker/README.md:3]: Could you please add more description here to explain the files in this repo and how people should build docker locally if needed?

Comment (geekflyer) [reply docker/README.md:3]: I think this is somewhat out of scope of this PR which just aims to just cleanup the old code.
I added a follow up ticket on #1412 . I'll work on this in conjunction with supporting M1 local builds.

Comment (davidiw) [reply docker/README.md:3]: I'm screenshotting this promise.

Comment (davidiw) [reply docker/README.md:3]: image

**Review by davidiw — APPROVED** [review]
/land

**Pull request closed without merging**